### PR TITLE
fix: Check values in strict cast Int to Time

### LINKED ISSUE #18844: Casting negative or large values to Time causes panic
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
>>> import polars as pl
>>> pl.Series([-1]).cast(pl.Time)
>>> pl.Series([86400000000000]).cast(pl.Time)
```

### Log output

```shell
NA
```

### Issue description

Values before 0:00 and after 24:00 are illegal for the Time type, but it appears that when casting Int32 to Time, the inclusion of these values causes panic.

### Expected behavior

Should not cause panic.

### Installed versions

<details>

```
--------Version info---------
Polars:              1.7.1
Index type:          UInt32
Platform:            Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:              3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
cloudpickle          <not installed>
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               <not installed>
gevent               <not installed>
great_tables         <not installed>
matplotlib           <not installed>
nest_asyncio         <not installed>
numpy                <not installed>
openpyxl             <not installed>
pandas               <not installed>
pyarrow              <not installed>
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                <not installed>
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by coastalwhite**
Fixes #18844.

### CODE REVIEW

File: crates/polars-core/src/chunked_array/logical/time.rs [hunk 1]
```diff
@@ -10,8 +13,47 @@ impl From<Int64Chunked> for TimeChunked {
...
 
 impl Int64Chunked {
-    pub fn into_time(self) -> TimeChunked {
-        TimeChunked::new_logical(self)
+    pub fn into_time(mut self) -> TimeChunked {
+        let mut null_count = 0;
+
+        let chunks = std::mem::take(&mut self.chunks)
+            .into_iter()
+            .map(|chunk| {
```
Comment (ritchie46) [inline crates/polars-core/src/chunked_array/logical/time.rs:21]: I think we can use `to_arrow` which gives us the logical arrow type and then call the arrow `cast` kernels. Those should do these checks.

Comment (coastalwhite) [reply crates/polars-core/src/chunked_array/logical/time.rs:21]: The `polars_arrow::compute::cast::cast` does not do these checks, if you are referring to that.

Comment (ritchie46) [reply crates/polars-core/src/chunked_array/logical/time.rs:21]: Then that kernel should.

Comment (coastalwhite) [reply crates/polars-core/src/chunked_array/logical/time.rs:21]: Okay, I will fix that then.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**